microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/screens/map/MapViewModel.kt

```diff
@@ -155,6 +163,23 @@ class MapViewModel @Inject constructor(
         )
         return Result.success(Unit)
     }
+    
+    fun selectNewFeature() {
+        require(_uiState.value is UIState.Switching)
+        val currentState = _uiState.value as UIState.Switching
+        currentState.oldState.featureForm.discardEdits()
+        val layer = currentState.oldState.featureForm.feature.featureTable?.layer as FeatureLayer
+        layer.clearSelection()
+        layer.selectFeature(currentState.newFeature)
+        _uiState.value =
+            UIState.Editing(featureForm = FeatureForm(currentState.newFeature, layer.featureFormDefinition!!))
+    }
+    
+    fun continueEditing() {
+        require(_uiState.value is UIState.Switching)
```

Comment:
```suggestion
        if(_uiState.value is UIState.Switching)
```
Similarly, here too.